src/scene/graph-node.js

```diff
@@ -911,12 +913,27 @@ Object.assign(pc, function () {
             }
         },
 
+        _unfreezeParentToRoot: function () {
+            var p = this._parent;
+            while (p) {
```

Comment:
Can we exit the loop here if `p._frozen === false`? That means we already have a path up to the root that is unfrozen?

Reply:
That was my initial idea, but unfortunately there is a case when after some cloning-reparenting operations sync becomes broken. I had to implement '_dirtifyWorldInternal' method to minimise performance impact.